### PR TITLE
Add order shipments resource

### PR DESCRIPTION
**Pull request by Iapetus-11**
Resolves EVIL-245

Adds order shipments resource and enums for shipping methods and providers

### CODE REVIEW

**Review by maxheyn — APPROVED** [review]
lgtm, can you provide an example showing the usage and that it works?

**Comment by Iapetus-11** [comment]
> lgtm, can you provide an example showing the usage and that it works?

Well I don't know how to test the creation, but I did run it and verify that I got a 400 because I put in intentionally wrong IDs. As for the all() method, I used the below script to test:
```py
from bigc import BigCommerceAPI
from bigc.exceptions import BadRequestError

store_hash = ''
access_token = ''

bigcommerce = BigCommerceAPI(store_hash, access_token)

orders = list(bigcommerce.orders.all(customer_id=1))
print(orders)

order_id = orders[0]['id']
print(f'{order_id=}')

shipments = list(bigcommerce.order_shipments.all(order_id))
print(f'{shipments=}')
bigcommerce.order_shipments.create(order_id, 123, [{'order_product_id': 69, 'product_id': 87, 'quantity': 420}])
```

**Review by r-thomson — CHANGES_REQUESTED** [review]
I'm inclined to say that these should be spelled `bigc.orders.all_shipments()` instead of `bigc.order_shipments.all()`. We're already following that pattern for, e.g., `orders.all_products()`.

**Comment by Iapetus-11** [comment]
> I'm inclined to say that these should be spelled `bigc.orders.all_shipments()` instead of `bigc.order_shipments.all()`. We're already following that pattern for, e.g., `orders.all_products()`.

🥱 

image

**Comment by Iapetus-11** [comment]
> > I'm inclined to say that these should be spelled `bigc.orders.all_shipments()` instead of `bigc.order_shipments.all()`. We're already following that pattern for, e.g., `orders.all_products()`.
> 
> 🥱
> 
> image

Ahh so you mean just stick it under orders, alright sure

**Review by maxheyn — APPROVED** [review]
shore

File: README.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ The following resources are currently supported:
 - `customer_groups`
 - `customers`
 - `orders`
+- `order_shipments`
```
Comment (r-thomson) [inline README.md:37]: This should be removed now
```suggestion
```

File: bigc/resources/orders.py [hunk 2]
```diff
@@ -72,3 +73,40 @@ def all_shipping_addresses(self, order_id: int) -> Iterator[dict]:
...
+            order_address_id: int,
+            items: list[dict],
+            *,
+            tracking_number: Optional[str] = None,
+            shipping_method: Optional[BigCommerceShippingMethod] = None,
+            shipping_provider: Optional[BigCommerceShippingProvider] = None,
+            tracking_carrier: Optional[str] = None,
+            comments: Optional[str] = None,
+            **kwargs,
+    ) -> dict:
```
Comment (r-thomson) [inline bigc/resources/orders.py:94]: IMO this is overcomplicating it. Trying to reproduce all of BigCommerce's fields in this library is just going to be way too much work to maintain.

Generally the pattern we have used so far is:

* Any "path" args (e.g. `order_id`)
* `*` to make sure everything else is a keyword arg
* Required fields (e.g. `order_address_id`)
* `**kwargs` for all optional fields

This is how I would expect this signature to look based on the docs:

```py
def create_shipment(self, order_id: int, *, order_address_id: int, items: list[dict], **kwargs) -> dict:
```

**Review by r-thomson — CHANGES_REQUESTED** [review]

File: bigc/data/shipping_method.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from enum import Enum, unique
+
+
+@unique
+class BigCommerceShippingMethod(Enum):
+    """Enumeration of BigCommerce shipping methods"""
+
+    STANDARD = 'Standard'
+    SHIP_BY_WEIGHT = 'Ship By Weight'
```
[the remark this answers is not in the document]
Comment (Iapetus-11) [reply bigc/data/shipping_method.py]: https://github.com/MedShift/bigc/pull/25/commits/bdfb8fb7cdc7d4338022155075cb983aa2a9af02

File: bigc/resources/orders.py (hunk 2 above)
Comment (Iapetus-11) [reply bigc/resources/orders.py]: https://github.com/MedShift/bigc/pull/25/commits/bdfb8fb7cdc7d4338022155075cb983aa2a9af02

File: README.md (hunk 1 above)
Comment (Iapetus-11) [reply README.md:37]: https://github.com/MedShift/bigc/pull/25/commits/bdfb8fb7cdc7d4338022155075cb983aa2a9af02

**Comment by maxheyn** [comment]
milo discovers the wonders of bigc PR standards :peepo:

**Review by CraigRobertWhite — APPROVED** [review]
lgtm

**Review by r-thomson — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (Iapetus-11) [reply README.md]: https://github.com/MedShift/bigc/pull/25/commits/d0a052f9c6ae3579134c25f508c84f2ab7f4079f

**Pull request merged**